doc/collision_contact_tutorial.rst

```diff
@@ -0,0 +1,201 @@
+Collision Contact Tutorial
+==========================
+
+This section walks you through C++ code which allows you to see collision contact points between the robot, itself, and the world as you move and interact with the robot’s arm in Rviz.
+
+Code
+----
+
+Classes
+^^^^^^^
+
+The code for this tutorial is mainly in the **InteractiveRobot** class which we will walk through below. The **InteractiveRobot** class maintains a RobotModel, a RobotState, and information about "the world" (in this case "the world" is a single yellow cube).
+
+The **InteractiveRobot** class uses the **IMarker** class which maintains an interactive marker. This tutorial does not cover the implementation of the IMarker class (`imarker.cpp <https://github.com/ros-planning/moveit_tutorials/blob/a0f75a9df840ac3dfd0eec337cea104a4c111747/doc/pr2_tutorials/interactivity/src/imarker.cpp>`_), but most of the code is copied from the `basic_controls <http://wiki.ros.org/rviz/Tutorials/Interactive%20Markers:%20Getting%20Started#basic_controls>`_ tutorial and you can read more there about interactive markers if you are interested.
+
+
+Code Description
+^^^^^^^^^^^^^^^^
+
+We will walk through the code in the order that it is run in the program, starting with the **main()** function in **collision_contact_tutorial.cpp**: 
+::
+      int main(int argc, char **argv)
```

Comment:
too much indent here, and throughout code samples below as visible on "View" page: https://github.com/0ldham/moveit_tutorials/blob/166e09203dc996df379de5f5dc4c01bf0963d6e0/doc/collision_contact_tutorial.rst

this is also giving an error on Travis CI